src/Cli/dotnet/ToolPackage/ToolPackageDownloader.cs

```diff
@@ -190,6 +197,52 @@ internal class ToolPackageDownloader : IToolPackageDownloader
                 });
         }
 
+        private static void CheckIfRequiredRuntimeIsInstalled(
+            ToolPackageInstance toolPackageInstance,
+            PackageId packageId,
+            bool isGlobalTool
+            )
+        {
+            var executableFilePath = toolPackageInstance.Commands[0].Executable;
+            var runtimeConfigFilePath = Path.ChangeExtension(executableFilePath.ToString(), ".runtimeconfig.json");
+
+            // Check if the runtimeconfig.json file is compatible with the current runtime
```

Comment:
Actually, I think this is a larger problem. Checking for compatibility with the loaded runtime means checking for frameworks - not just the core runtime. Since `dotnet.dll` only depends on the core runtime, it will only have that loaded.

Tools could depend on other frameworks like AspNetCore or WindowsDesktop, right? I expect any tools like that would fail the compatibility check here.

Reply:
Yes, tool could be depend on other frameworks.

Reply:
The new `hostfxr_resolve_frameworks_for_runtime_config` API considers other types of frameworks. We should call that and see if it returns 0.